src/main/java/org/cbio/gdcpipeline/writer/ClinicalPatientDataWriter.java

```diff
@@ -0,0 +1,179 @@
+package org.cbio.gdcpipeline.writer;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.cbio.gdcpipeline.model.cbio.CBioClinicalDataModel;
+import org.cbio.gdcpipeline.model.cbio.PatientFileModel;
+import org.cbio.gdcpipeline.util.ClinicalPatientFileHeaders;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.item.ItemStreamException;
+import org.springframework.batch.item.file.FlatFileHeaderCallback;
+import org.springframework.batch.item.file.FlatFileItemWriter;
+import org.springframework.batch.item.file.ResourceAwareItemWriterItemStream;
+import org.springframework.batch.item.file.transform.BeanWrapperFieldExtractor;
+import org.springframework.batch.item.file.transform.DelimitedLineAggregator;
+import org.springframework.batch.item.file.transform.FieldExtractor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by Dixit on 14/06/17.
+ */
+public class ClinicalPatientDataWriter implements ResourceAwareItemWriterItemStream<CBioClinicalDataModel> {
+
+    @Value("#{jobParameters[sourceDirectory]}")
+    private String sourceDir;
+
+    @Value("#{jobParameters[outputDirectory]}")
+    private String outputDir;
+
+    @Value("#{jobParameters[study]}")
+    private String cancer_study_id;
+
+    private String dataOutputFile;
+
+    private Resource resource;
+
+
+    @Value("${clinical.data.patient.file.name}")
+    private String CLINICAL_DATA_PATIENT_FILE_NAME;
```

Comment:
You could have just one writer that had both the patient and sample file names injected. It would select which file to write based on if the model being passed in was an instance of Patient or Sample.